packages/task/src/browser/task-service.ts

```diff
@@ -496,11 +501,14 @@ export class TaskService implements TaskConfigurationClient {
      * @param token  The cache token for the user interaction in progress
      */
     async runLastTask(token: number): Promise<TaskInfo | undefined> {
-        if (!this.lastTask) {
+        if (!this.lastTask || !this.lastTask.resolvedTask) {
             return;
         }
-        const { source, taskLabel, scope } = this.lastTask;
-        return this.run(token, source, taskLabel, scope);
+        if (!this.lastTask.resolvedTask.runOptions?.reevaluateOnRerun) {
```

Comment:
> If this flag is set to false no variables will be resolved again. Therefore lastTask can now be re-triggered without resolving all variables.

It appears that `reevaluateOnRerun` is being treated as the default. That's a change in the expectations for internal tasks, which in the past would always have been resolved when rerun. Is there any value in treating internal and external tasks differently in this respect?

Reply:
@colin-grant-work by "internal", do you mean tasks defined in `tasks.json` or contributed by non-plugin API? If we need to treat tasks different, I would much prefer putting the difference where we _create_ the tasks (for example by setting a flag in the contructor) and not where we process the tasks. Locality of reference for teh win!

Reply:
Re: `tasks.json`, we should likely follow what VSCode does, though there's scope for making our own choice there. Mostly I mean other tasks created internally, but I'm not very familiar with the behavior or UI of 'rerun' tasks (as opposed to running the same task again, but not as a rerun), so I'm not sure how disruptive it would be to change from resolving to not resolving variables in the task description.

 One solution might be to modify the plugin system to supply `false` by default for tasks originating from plugins, and then have a flag in the task service for whether to use `false` or `true` as the default value there so that application implementers can easily toggle it if they have a use case where `true` is appropriate. On the other hand, it sounds like @tsmaeder favors something on the task description itself (in which case maybe setting `false` in the plugin system and using `true` as a default otherwise would be the best course?). In any case, I don't know if it really warrants too much concern.

Reply:
I thought about this some more: what I _really_ would like to avoid is a "if tasksource = plugin, do x, else do y". I'm also not a fan of setting variables that are client API in order to simulate the "right" default behaviour: if anything, we should set a flag : `task.defaultRerunOnReevaluate= false` that is _not_ part of the API in the plugin task provider implementation and default to true in clients to preserve the current behaviour. 
But t.b.h, I would just make the change to the VS Code behavior: it's a breaking change, and while we try not to frivolously break API, we also should not unduly complicate the system to keep the API stable.
My vote is: put the code in as it is and make a note of it in the "breaking changes". Clients can trivially update their code if they rely on the behaviour.

Reply:
I'm fine with this approach.